### ISSUE TITLE
Wrong loop timing due to asynchronous execution

### ISSUE THREAD

**Issue by clampert** [louiswang524/jaxGPT#1]
Great repository, thank you for releasing it! Unfortunately, I suspect that the loop timing (and as a consequence the token thoughput) is incorrect, because jax runs the training steps asynchronously, see https://docs.jax.dev/en/latest/async_dispatch.html .
On RTX4090 with jax 0.6.2, the reported time per step is ~10ms, but only 2.3 iterations are completed per second. Inserting `jax.block_until_ready(loss)` before the second call to `time.time()` keeps the overall speed the same, but outputs the actual value of ~420ms per step.
